src/network/jsonwebtask.cpp

```diff
@@ -62,12 +62,33 @@ bool readJsonContent(
     DEBUG_ASSERT(jsonContent);
     DEBUG_ASSERT(JSON_MIME_TYPE.isValid());
     const auto contentType = readContentType(reply);
-    if (contentType == JSON_MIME_TYPE) {
-        *jsonContent = QJsonDocument::fromJson(reply->readAll());
-        return true;
-    } else {
+    if (contentType != JSON_MIME_TYPE) {
+        kLogger.warning()
+                << "Unexpected content type"
+                << contentType;
+        return false;
+    }
+    QByteArray jsonData = reply->readAll();
+    if (jsonData.isEmpty()) {
+        kLogger.warning()
+                << "Empty reply";
         return false;
     }
+    QJsonParseError parseError;
+    const auto jsonDoc = QJsonDocument::fromJson(
+            jsonData,
+            &parseError);
+    if (jsonDoc.isNull()) {
+        DEBUG_ASSERT(parseError.error != QJsonParseError::NoError);
```

Comment:
`null` is a complete, valid json document. The debug assertion won't trigger in that case, right?

Reply:
`QByteArray()` -> `QJsonParseError::IllegalValue` (both null and empty)
`""` -> `QJsonParseError::IllegalValue` (both null and empty)
`" "` -> `QJsonParseError::IllegalValue` (both null and empty)
`"null"` -> `QJsonParseError::IllegalValue` (both null and empty)
`"{}"` -> `QJsonParseError::NoError` (not null, not empty, i.e. a single empty object)

Any ideas how to handle this correctly?

Reply:
https://doc.qt.io/qt-5/qjsondocument.html#fromJson
> Returns a valid (non-null) QJsonDocument if the parsing succeeds. If it fails, the returned document will be null, and the optional error variable will contain further details about the error.

So the parse error might only be assigned if the returned document is null. Otherwise we should not evaluate it. I will add a comment, just in case.